### DISCUSSION TITLE
How to show WhatsApp QR code in the Terminal on WSL?

### DISCUSSION THREAD

**Discussion by ammos** [d99kris/nchat#118] [Q&A]
I have a problem with QR code in terminal. I cannot see it. How to fix it?

**Comment by d99kris** [comment]
Hi @ammos - which OS is this on?

`nchat` uses `xdg-open` on Linux to display an image of the QR code if environment variable `DISPLAY` is set. On macOS it always uses an image viewer.

I believe one can force `nchat` to display the QR code in the terminal by doing something like:
```
unset DISPLAY
nchat -s
```
If it still does not display it, please check `~/.nchat/log.txt` for any warning or error messages that might give some hint.

On a related note, I'll be looking into whether there's possibility of non-QR-code setup. I saw the underlying library whatsmeow recently added capabilities for something that seems like that in https://github.com/tulir/whatsmeow/issues/419 but it's not yet integrated into nchat, and I'm not sure it will allow text-only setup, I'll need to confirm/test that.

**Comment by ammos** [comment]
I am using Ubuntu 22.04 under WSL.

```
unset DISPLAY
nchat -s
```

Thank you, it worked!

It’s a pity that man can’t combine telegram and whatsapp.

**Reply by d99kris** [reply]
Good to hear it worked, I'll look into making the X/DISPLAY check more robust. I did test it under WSL a while back, but maybe things have changed. 

As for the combined Telegram/WhatsApp, I might look into officially supporting it, but currently it's not. See #102 for more details.
